platform/lib/routers/surveyComponent.js

```diff
@@ -0,0 +1,131 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const Joi = require('joi');
+const express = require('express');
+const log = require('@lib/utils/log')('Survey Component Response');
+const {GoogleSpreadsheet} = require('google-spreadsheet');
+const Doc = new GoogleSpreadsheet(process.env.SURVEY_RESPONSE_SHEET_ID);
+const surveyEndpoint = '/fezSurveyResponse';
+
+function validateRequest(req, res, next) {
+  if (req.method !== 'POST') {
+    return res.sendStatus(405);
+  }
+
+  const schema = Joi.object({
```

Comment:
Does this need to be created on every request?

Reply:
great catch, no. extracted it out